src/pkg/cli/composeUp.go

```diff
@@ -96,6 +97,13 @@ func ComposeUp(ctx context.Context, loader client.Loader, c client.FabricClient,
 			}
 			deployRequest.DelegationSetId = delegation.DelegationSetId
 		}
+		go func() {
+			<-ctx.Done()
+
+			if errors.Is(ctx.Err(), context.Canceled) || errors.Is(ctx.Err(), context.DeadlineExceeded) {
```

Comment:
This is not enough to exclude the natural cancellation of the context from success :(

```
defang up
 * Using Defang Playground provider from stored preference
 * Packaging the project files for app at /Users/jordan/wk/samples/samples/nodejs-http/app
 * Uploading the project files for app
 * Monitor your services' status in the defang portal
   - https://portal.defang.dev/service/app
 * Tailing logs for deployment ID qqut567fztmx ; press Ctrl+C to detach:
 * Showing only build logs and runtime errors. Press V to toggle verbose mode.
2024-11-19T15:52:44.499-08:00 cd Update started for stack jordanstephens-prod1
2024-11-19T15:52:48.671-08:00 cd  ** Building image for "app"...
2024-11-19T15:52:51.878-08:00 cd  ** Build succeeded for "app"
2024-11-19T15:52:52.356-08:00 cd  ** Updated service "app" to revision 50
2024-11-19T15:52:56.752-08:00 cd Update succeeded in 12.284256189s ; provisioning...
2024-11-19T15:53:39.389-08:00 app Server running at http://127.0.0.1:3000/
errCompleted
 ! Deployment will not be cancelled.
 * Service app is in state DEPLOYMENT_COMPLETED and will be available at:
   - https://jordanstephens-app--3000.prod1a.defang.dev
 * Done.
For help with warnings, check our FAQ at https://docs.defang.io/docs/faq
```

Reply:
So does that mean this PR is not ready to be merged? I wouldn't want the happy path to show "Deployment will not be cancelled".

Reply:
Right this PR is not ready to merge. It's still in draft.